rest/server_context.go

```diff
@@ -946,6 +954,92 @@ func (sc *ServerContext) getDbConfigFromServer(dbName string) (*DbConfig, error)
 	return &config, nil
 }
 
+//////// STATS LOGGING
+
+func (sc *ServerContext) startStatsLogger() {
+
+	statsLogFrequencySecs := 5
+	if sc.config.Unsupported != nil && sc.config.Unsupported.Logging != nil {
+		statsLogFrequencySecs = sc.config.Unsupported.Logging.StatsLogFrequencySecs
+	}
+
+	interval := time.Second * time.Duration(statsLogFrequencySecs)
+
+	sc.statsLoggingTicker = time.NewTicker(interval)
+	go func() {
+		for range sc.statsLoggingTicker.C {
+			sc.logStats()
+		}
+	}()
+	base.Infof(base.KeyAll, "Logging stats with frequency: %v", interval)
+
+}
+
+func (sc *ServerContext) stopStatsLogger() {
+	if sc.statsLoggingTicker != nil {
+		sc.statsLoggingTicker.Stop()
+	}
+}
+
+func (sc *ServerContext) logStats() {
+
+	AddGoRuntimeStats()
+
+	// Create wrapper expvar map in order to add a timestamp field for logging purposes
+	logStats := new(expvar.Map).Init()
+	logStats.Set("stats", base.Stats)
+	logStats.Set("unix_epoch_timestamp", base.ExpvarInt64Val(time.Now().Unix()))
+
+	// Marshal expvar map w/ timestamp to string and write to logs
+	base.Statsf([]byte(logStats.String()))
+
+}
+
+func AddGoRuntimeStats() {
+
+	statsResourceUtilization := base.StatsResourceUtilization()
+
+	// Num goroutines
+	statsResourceUtilization.Set("num_goroutines", base.ExpvarIntVal(runtime.NumGoroutine()))
+
+	// Read memstats (relatively expensive)
+	memstats := runtime.MemStats{}
+	runtime.ReadMemStats(&memstats)
+
+	// memory_rss (rss = resident set size)
+	// Calculate this the same way that FTS does.
+	// TODO: document why this works (blog post or email thread)
+	// TODO: according to the PRD, this should figure out the total memory in the system and express as a ratio of total memory
+	memory_rss := int(memstats.Sys - memstats.HeapReleased) // convert uint -> int since that's what expvar supports
+	statsResourceUtilization.Set("memory_rss_bytes", base.ExpvarIntVal(memory_rss))
+
+	// Sys
+	statsResourceUtilization.Set("go_memstats_sys", base.ExpvarIntVal(int(memstats.Sys)))
+
+	// HeapAlloc
+	statsResourceUtilization.Set("go_memstats_heapalloc", base.ExpvarIntVal(int(memstats.HeapAlloc)))
+
+	// HeapIdle
+	statsResourceUtilization.Set("go_memstats_heapidle", base.ExpvarIntVal(int(memstats.HeapIdle)))
+
+	// HeapInuse
+	statsResourceUtilization.Set("go_memstats_heapinuse", base.ExpvarIntVal(int(memstats.HeapInuse)))
+
+	// HeapReleased
+	statsResourceUtilization.Set("go_memstats_heapreleased", base.ExpvarIntVal(int(memstats.HeapReleased)))
+
+	// StackInuse
+	statsResourceUtilization.Set("go_memstats_stackinuse", base.ExpvarIntVal(int(memstats.StackInuse)))
+
+	// StackSys
+	statsResourceUtilization.Set("go_memstats_stacksys", base.ExpvarIntVal(int(memstats.StackSys)))
+
+	// PauseTotalNs
+	statsResourceUtilization.Set("go_memstats_pausetotalns", base.ExpvarIntVal(int(memstats.PauseTotalNs)))
```

Comment:
For all of these we could be using int64 instead to avoid the double type conversion `uint64->int->int64->Val` inside ExpvarIntVal.

Reply:
Fixed by adding a type-specific method for `uint64` expvar var conversion